### PR TITLE
GUP-222 fixed heartbeat interval

### PR DESCRIPTION
**Pull request by mamachanko**
Rather than scheduling the next heartbeat dynamically we fix it to a value. Includes some refactoring. Removes Unicode character from logs. Heartbeat is virtually `ping`. Heartbeat is used to measure `rtt`.

### CODE REVIEW

File: lymph/core/connection.py [hunk 1]
```diff
@@ -74,35 +69,39 @@ def heartbeat_loop(self):
...
+                self.set_status(IDLE)
+                self.idle_since = now
+            else:
+                self.set_status(RESPONSIVE)
+
+    def log_stats(self):
+        roundtrip_stats = 'window (mean rtt={mean:.1f} ms; stddev rtt={stddev:.1f})'.format(**self.heartbeat_samples.stats)
+        roundtrip_total_stats = 'total (mean rtt={mean:.1f} ms; stddev rtt={stddev:.1f})'.format(**self.heartbeat_samples.total.stats)
+        logger.debug("pid=%s; %s; %s; phi=%.3f; ping/s=%.2f; status=%s" % (
+            os.getpid(),
```
Comment (mouadino) [inline lymph/core/connection.py]: can we memoize this ? syscall are not cheap.

File: lymph/core/connection.py [hunk 2]
```diff
@@ -74,35 +69,48 @@ def heartbeat_loop(self):
...
+            self.pid,
+            roundtrip_stats,
+            roundtrip_total_stats,
+            self.phi,
+            self.explicit_heartbeat_count / max(1, time.monotonic() - self.created_at),
+            self.status,
+        ))
+
+    @property
+    def pid(self):
```
Comment (mouadino) [inline lymph/core/connection.py]: Can we just set it in ``__init__`` ?

Comment (mamachanko) [inline lymph/core/connection.py]: silly me

**Comment by mouadino** [comment]
LGTM

probably @emulbreh should take merge this one :)

File: lymph/core/connection.py [hunk 3]
```diff
@@ -49,6 +44,8 @@ def __init__(self, server, endpoint, heartbeat_interval=1, timeout=1, idle_timeo
         self.heartbeat_loop_greenlet = self.server.container.spawn(self.heartbeat_loop)
         self.live_check_loop_greenlet = self.server.container.spawn(self.live_check_loop)
 
+        self.pid = os.getpid()
```
Comment (emulbreh) [inline lymph/core/connection.py]: We can just remove the pid from the logs.

**Pull request merged**